packages/flutter_tools/lib/src/macos/cocoapod_utils.dart

```diff
@@ -21,7 +21,7 @@ Future<void> processPodsIfNeeded(
   final FlutterProject project = xcodeProject.parent;
   // Ensure that the plugin list is up to date, since hasPlugins relies on it.
   await refreshPluginsList(project);
-  if (!(hasPlugins(project) || (project.isModule && xcodeProject.podfile.existsSync()))) {
+  if (!hasPlugins(project) && !xcodeProject.podfile.existsSync()) {
```

Comment:
This is the part that should fix the post-commit failure from https://github.com/flutter/flutter/pull/64674.
@xster What do you think about this?  Looks like the `project.isModule` caveat was added back in #21276.

Reply:
nit. Maybe a comment to explain why we have to `pod install` even if it doesn't have plugins?

Reply:
Good idea, done.

Reply:
I wish that original line could be DeMorgan rewritten 😂

That isModule part is definitely weird. I can see the argument for just generally not running pod install when there's only a user Podfile without plugin. Since at that point, it doesn't have much to do with Flutter's build pipeline. When a user changes his own Podfile, the user should run pod install themselves. Our customers did complain back in beta days about pod install always running for no changes and being slow. 

I can't think of why isModule + has Podfile means pod instead has to run. Removing isModule from this consideration altogether seems reasonable. "'Cause if you like it, then you shoulda put a test on it"

Reply:
I reverted this, and instead added a Flutter plugin to the integration test so the tool runs `pod install` for us and I don't need to call it manually.